app/models/concerns/location_feed.rb

```diff
@@ -0,0 +1,16 @@
+module LocationFeed
+  def for_feed
+    super.tap do |common_feed_attributes|
+      location_key = self.class::LOCATION_ATTRIBUTE_KEY.to_s
+      location_method = location_key.sub('_id', '')
+      prefix = location_key == 'location_id' ? nil : location_key.sub('_location_id', '')
+
+      feed_details = common_feed_attributes['details']
+        .deep_dup
+        .except(location_key)
+      feed_details.merge!(public_send(location_method).for_feed(prefix: prefix))
+
+      common_feed_attributes['details'] = feed_details
```

Comment:
This cloning and merging stuff seems slight overkill - could the `details` part not be updated in situ instead?

```suggestion
      updated_location_mapping = public_send(location_method).for_feed(prefix: prefix))
      common_feed_attributes['details'].delete(location_key).merge!(updated_location_mapping)
```

or perhaps the `delete` isn't even needed as the same key is being overwritten (I think).. so

```suggestion
      updated_location_mapping = public_send(location_method).for_feed(prefix: prefix))
      common_feed_attributes['details'].merge!(updated_location_mapping)
```

The name `updated_location_mapping` may need some thought - I'm assuming this returns a key/value pair. This feels clearer to follow to me (if it works!).

Reply:
Looks nicer now, thanks for cleaning up 👍